qiskit_nature/problems/sampling/protein_folding/interactions/interaction.py

```diff
@@ -0,0 +1,34 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An abstract class defining an interaction between beads of a peptide."""
+from abc import ABC, abstractmethod
+from typing import List
+import numpy as np
+
+
+class Interaction(ABC):
+    """An abstract class defining an interaction between beads of a peptide."""
+
+    @abstractmethod
+    def calc_energy_matrix(self, chain_len: int, residue_sequence: List[str]) -> np.ndarray:
```

Comment:
@dlasecki seems that `calc_energy_matrix ` is breaking the sphinx spell check. Should we update the pylint dictionary or rename this function to `calculate_energy_matrix` ?

Reply:
It is weird cause that name did not change and was fine for the check before...

Reply:
You are right. Private methods are not triggering this sphinx spelling check problem.

Reply:
Private methods do not end up in the API ref (i.e in the resultant html)